### PR TITLE
tests: add tests for `zkfs` utilities

### PR DESCRIPTION
**Pull request by EshaanAgg**
## Description
- Added the tests for some utilities for `vitess/go/cmd/zk/internal/zkfs`
- I couldn't add the complete coverage code as I am having trouble mocking the `zk2topo.ZkConn` in a simple manner for testing. If anyone can point me to any files that might have done the same previously, or give me some clue as to how we might go about it (without defining all the required methods explicitly), I would love to update the PR to handle the same as well. 

## Related Issue(s)
Fixes part of #14931

### CODE REVIEW

File: go/cmd/zk/internal/zkfs/zkfs_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/z-division/go-zookeeper/zk"
+)
+
+func TestFS_CopyContext(t *testing.T) {
+	// fs := &FS{}
```
Comment (GuptaManan100) [inline go/cmd/zk/internal/zkfs/zkfs_test.go:28]: Is this a testing artifact? Do you want to remove this in the final PR?

File: go/cmd/zk/internal/zkfs/zkfs_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+	// Create a temporary directories for testing
+	srcPath, err := os.MkdirTemp("", "source")
+	assert.NoError(t, err)
+	defer os.RemoveAll(srcPath)
+
+	dstPath, err := os.MkdirTemp("", "destination")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dstPath)
```
Comment (GuptaManan100) [inline go/cmd/zk/internal/zkfs/zkfs_test.go:37]: This test isn't doing anything, other than just creating and deleting temporary folders.

File: go/cmd/zk/internal/zkfs/zkfs_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+func TestParsePermMode(t *testing.T) {
+	assert.Equal(t, int32(0), ParsePermMode("zk"))
+	assert.Equal(t, int32(zk.PermRead|zk.PermWrite), ParsePermMode("zkrw"))
+	assert.Equal(t, int32(zk.PermRead|zk.PermWrite|zk.PermAdmin), ParsePermMode("zkrwa"))
+	assert.PanicsWithValue(t, "invalid mode", func() {
+		ParsePermMode("")
+		ParsePermMode("z")
+	})
```
Comment (GuptaManan100) [inline go/cmd/zk/internal/zkfs/zkfs_test.go:54]: The two `ParsePermMode()` calls should be in separate assert calls. Currently code is testing that either one of them panics, but I think you want to test that both panic.

File: go/cmd/zk/internal/zkfs/zkfs_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		},
+		{
+			name:     "Mixed Permissions",
+			acl:      zk.ACL{Perms: zk.PermRead | zk.PermDelete | zk.PermAdmin},
+			expected: "r-d-a",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (ajm188) [inline go/cmd/zk/internal/zkfs/zkfs_test.go:91]: you need to capture the iterator within the loop scope here, see https://stackoverflow.com/a/33459174 for an example/explainer

File: go/cmd/zk/internal/zkfs/zkfs_test.go (hunk 1 above)
Comment (EshaanAgg) [reply go/cmd/zk/internal/zkfs/zkfs_test.go:28]: Yes my bad. Will do.

File: go/cmd/zk/internal/zkfs/zkfs_test.go (hunk 2 above)
Comment (EshaanAgg) [reply go/cmd/zk/internal/zkfs/zkfs_test.go:37]: Yes. Committed the same by mistake.

File: go/cmd/zk/internal/zkfs/zkfs_test.go (hunk 3 above)
Comment (EshaanAgg) [reply go/cmd/zk/internal/zkfs/zkfs_test.go:54]: Nice catch! Thanks!

File: go/cmd/zk/internal/zkfs/zkfs_test.go (hunk 4 above)
Comment (EshaanAgg) [reply go/cmd/zk/internal/zkfs/zkfs_test.go]: Got it!

Comment (EshaanAgg) [reply go/cmd/zk/internal/zkfs/zkfs_test.go]: Do we really need this? Aren't both the `tc` the same? Isn't capturing the scope of the iterator something we need to do when we are starting different goroutines in the for loop (like `go t.Run(tc.name, ......)`?

Comment (ajm188) [reply go/cmd/zk/internal/zkfs/zkfs_test.go]: Yes, but, it is safer to write it this way, in case we ever go back and make these parallel

Comment (EshaanAgg) [reply go/cmd/zk/internal/zkfs/zkfs_test.go]: Got it.

**Review by ajm188 — APPROVED** [review]

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**